type_info.go

```diff
@@ -232,6 +235,25 @@ func NewMapInfo(keyInfo TypeInfo, valueInfo TypeInfo) (TypeInfo, error) {
 	return info, nil
 }
 
+// NewArrayInfo returns ARRAY type information.
+// childInfo contains the type information of the ARRAY's elements.
+// size is the ARRAY's fixed size.
+func NewArrayInfo(childInfo TypeInfo, size uint64) (TypeInfo, error) {
+	if childInfo == nil {
+		return nil, getError(errAPI, interfaceIsNilError("childInfo"))
+	}
+	if size == 0 {
```

Comment:
Edge case, but I think there can also be an array of size 0 in Go, even if this makes no sense.

Reply:
This is a duckdb restriction - so I propagated that to Go. 
```
D CREATE TABLE tbl (i INT[0]);
Parser Error: Arrays must have a size of at least 1
```